### PR TITLE
Bug DCOS-9904: Patch to change navstar overlay polling logic

### PR DESCRIPTION
**Pull request by GoelDeepak**
Two changes:
1. Navstar backs off if the last poll request was successful
2. Navstar apply changes only if there is an event from lashup

### CODE REVIEW

File: apps/navstar_overlay/src/navstar_overlay_poller.erl [hunk 1]
```diff
@@ -185,6 +194,14 @@ scheme() ->
             "http"
     end.
 
+update_poll_period(OldPollPeriod) ->
+    case (NewPollPeriod = OldPollPeriod * 2) > ?MAX_POLL_PERIOD of
```
Comment (sargun) [inline apps/navstar_overlay/src/navstar_overlay_poller.erl]: Please break up the initial part of the statement and the later part of the statement. Alternatively, use a guard on the update_poll_period function itself.

File: apps/navstar_overlay/src/navstar_overlay_poller.erl [hunk 2]
```diff
@@ -355,23 +372,24 @@ vtep_mac(IntList) ->
...
-        fun(Entry, Acc) -> maybe_configure_overlay_entry(Overlay, Entry, Acc) end,
+        fun(Entry, SubAcc) -> try_configure_overlay(Entry, Subnet, State0, LashupValue, SubAcc) end,
         State0,
-        LashupValue
-    ).
+        KnownOverlays).
+
+try_configure_overlay(Overlay, Subnet, State0, LashupValue, _Acc) ->
+    #mesos_state_agentoverlayinfo{info = #mesos_state_overlayinfo{subnet = OverlaySubnet}} = Overlay,
+    case Subnet == OverlaySubnet of
```
Comment (sargun) [inline apps/navstar_overlay/src/navstar_overlay_poller.erl]: Use a guard on the function.

File: apps/navstar_overlay/src/navstar_overlay_poller.erl [hunk 3]
```diff
@@ -355,23 +372,24 @@ vtep_mac(IntList) ->
...
+        true -> lists:foldl(
```
Comment (sargun) [inline apps/navstar_overlay/src/navstar_overlay_poller.erl]: the list:foldl should be on the next line.

File: apps/navstar_overlay/src/navstar_overlay_poller.erl [hunk 4]
```diff
@@ -29,14 +29,19 @@
 
 -define(SERVER, ?MODULE).
 -define(POLL_PERIOD, 30000).
```
Comment (sargun) [inline apps/navstar_overlay/src/navstar_overlay_poller.erl]: Maybe MIN_POLL_PERIOD?

**Review by sargun — CHANGES_REQUESTED** [review]

**Comment by GoelDeepak** [comment]
@sargun I have incorporated your review comments. DCOS integration tests are passing with these changes. I have also verified that:
1. Navstar listens and acts on lashup_kv events, 
2.  Navstar backs off on successful last poll

**Pull request merged**